library/Icingadb/Web/Navigation/Action.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+/* Icinga DB Web | (c) 2021 Icinga GmbH | GPLv2 */
+
+namespace Icinga\Module\Icingadb\Web\Navigation;
+
+use Icinga\Module\Icingadb\Common\Macro;
+use Icinga\Module\Icingadb\Compat\CompatHost;
+use Icinga\Module\Icingadb\Compat\CompatService;
+use Icinga\Web\Navigation\NavigationItem;
+use ipl\Stdlib\Filter;
+use ipl\Web\Filter\QueryString;
+use ipl\Web\Url;
+
+class Action extends NavigationItem
+{
+    use Macro;
+
+    /**
+     * Whether this action's macros were already resolved
+     *
+     * @var bool
+     */
+    protected $resolved = false;
+
+    /**
+     * This action's object
+     *
+     * @var CompatHost|CompatService
+     */
+    protected $object;
+
+    /**
+     * The filter to use when being asked whether to render this action
+     *
+     * @var string
+     */
+    protected $filter;
+
+    /**
+     * This action's raw url attribute
+     *
+     * @var string
+     */
+    protected $rawUrl;
+
+    /**
+     * Set this action's object
+     *
+     * @param  CompatHost|CompatService $object
+     *
+     * @return $this
+     */
+    public function setObject($object)
+    {
+        $this->object = $object;
+
+        return $this;
+    }
+
+    /**
+     * Get this action's object
+     *
+     * @return CompatHost|CompatService
+     */
+    public function getObject()
+    {
+        return $this->object;
+    }
+
+    /**
+     * Set the filter to use when being asked whether to render this action
+     *
+     * @param   string  $filter
+     *
+     * @return  $this
+     */
+    public function setFilter($filter)
+    {
+        $this->filter = $filter;
+
+        return $this;
+    }
+
+    /**
+     * Get the filter to use when being asked whether to render this action
+     *
+     * @return  string
+     */
+    public function getFilter()
+    {
+        return $this->filter;
+    }
+
+    /**
+     * Set this item's url
+     *
+     * @param \Icinga\Web\Url|string $url
+     *
+     * @return $this
+     */
+    public function setUrl($url)
+    {
+        if (is_string($url)) {
+            $this->rawUrl = $url;
+        } else {
+            parent::setUrl($url);
+        }
+
+        return $this;
+    }
+
+    /**
+     * @inheritDoc
+     */
```

Comment:
Please don't use this. Only use it if you combine it with custom documentation.

This is fine:

```
/**
* Returns  my super custom type
*
* @inheritdoc
*
* @return SuperCustomType
* /
```